Main/Source/game.cpp

```diff
@@ -3270,50 +3277,93 @@ int game::Load(cfestring& saveName)
   return LOADED;
 }
 
-festring game::SaveName(cfestring& Base)
+/**
+ * this prevents all possibly troublesome characters in all OSs
+ */
+void fixChars(festring& fs)
 {
-  festring SaveName = GetSaveDir();
+  for(festring::sizetype i = 0; i < fs.GetSize(); ++i)
+  {
+    if(fs[i]>='A' && fs[i]<='Z')continue;
+    if(fs[i]>='a' && fs[i]<='z')continue;
+    if(fs[i]>='0' && fs[i]<='9')continue;
+
+    fs[i] = '_';
+  }
+}
```

Comment:
Really hope #433 will be reviewed and merged soon. `fs[i] = X` is still harmful.

Reply:
I will take a look on it thx

Reply:
Without my PR, you can create a local temporary festring, manipulate the temp string and then `fs.empty(); fs << temp;` If my PR get merged first, then you don't have to change anything.

`festring::sizetype` is unrelated to the size of `char`. "character/glyph" and `char` are different things. It can already work for all language if the encoding of input text does not conflict with this single byte manipulation of ASCII text.